src/fb.c

```diff
@@ -40,24 +43,147 @@ static inline uint32_t rgbx_to_rgb30(u32 c)
     return (b << 2) | (g << 12) | (r << 22);
 }
 
+static void fb_set_pixel(int x, int y, u32 val)
+{
+    fb.ptr[x + y * fb.s] = val;
+}
+
+static u32 fb_console_get_row(u32 row)
+{
+    return (row + console.row_offset) % CONSOLE_MAX_ROWS;
+}
+
+static u32 font_get_pixel(u8 c, u32 x, u32 y)
+{
+    c -= 0x20;
+    u8 *ptr = &_binary_font_b_start[c * 512 + (y * 8 + x) * 4];
+    u8 r = *ptr++;
+    u8 g = *ptr++;
+    u8 b = *ptr++;
+
+    return (b << 2) | (g << 12) | (r << 22);
```

Comment:
You need to use a memcpy to do that safely but otherwise I think that's correct.

Reply:
There are other place in this code that does this. So I assume HW take care of unaligned access. I agree that this way is safer. Maybe fix it for fb_blit() as well so it consistent?

Reply:
Why is a memcpy required here?

Reply:
Yeah the latter. Just for aliasing, since even in C you can't just use a pointer cast legally to do that (a union can work but the details are confusing). The bootargs struct is already done like this, other code seems to forget, but it of course should be optimized into a pure load. We could also compile with "-fno-strict-aliasing" which is what linux does but yeah. I hate this issue because I feel like the cast is so much clearer.